set.gemspec

```diff
@@ -1,6 +1,13 @@
+name = File.basename(__FILE__, ".gemspec")
+version = ["lib", Array.new(name.count("-")+1, "..").join("/")].find do |dir|
```

Comment:
I understand this is a generalized piece of code, but maybe I'll fix it to look more readable.